### ISSUE TITLE
Increased latency when using ffmpeg to restream via srt to an rtp destination

### ISSUE THREAD

**Issue by marcelgoya** [Haivision/srt#544]
Hi,

I'm trying to restream an srt input stream to a rtp destination via ffmpeg:

`ffmpeg -i srt://:1234?mode=listener -c:v copy -vtag avc1 -deadline realtime -preset ultrafast -tune zerolatency -an -f rtp rtp://127.0.0.1:8004`

Unfortunately, I've noticed that there's a latency of around 1-2 seconds when I'm trying to send the stream via srt. There is almost no latency when I'm sending the stream directly via rtp. 

My question is, do you know if it could be a problem of the srt implementation inside ffmpeg or could it be a missing flag that I'm not aware of?

Thanks in advance!

**Comment by maxlovic** [comment]
Hi @marcelgoya,
We are aware of some issues in SRT implementation inside `ffmpeg`, but it is hard to tell what you are facing.

You have SRT listener in both cases, right? One case is you receive over SRT and stream over SRT. Another case (the one you've shown above) is receive over SRT and transmit over RTP. And the second one works better. Is it correct?

**Comment by marcelgoya** [comment]
Hi @maxlovic,

I'd figured this problem would be hard to track down :) Basically, the stream is forwarded in the following order:

WebRTC input -> Wowza Engine -> restream to destination server via SRT and converting the SRT input via FFmpeg to local RTP port (Janus server). 

I'm experiencing a lag of around 1-2 seconds when using the combination from above. I've also tried doing the following:

WebRTC input -> Wowza Engine -> restream to destination server via RTP directly to Janus server

Sending the stream directly without ffmpeg SRT command removes the lag and stream is viewable in real-time. Therefore, I assume it must be some missing param inside the ffmpeg command that does some sort of buffering or something else that creates this lag.

Cheers,
Marcel

**Comment by maxlovic** [comment]
@marcelgoya 
Can you measure the latency on the ffmpeg just playing back the stream. E.g. do smth like `ffplay srt://:1234?mode=listener`?
So far it does look like like some internal buffering in ffmpeg pipeline.

**Comment by marcelgoya** [comment]
@maxlovic Sure, I'll get back to you once that's done. Yes, that's exactly what I'm assuming ....

**Comment by marcelgoya** [comment]
@maxlovic Alright, I did some more thorough testing and came to the conclusion that the problem wasn't related to ffmpeg at all. I used the the srt-live-transmit program and watched the live stream via VLC and noticed that the delay was caused by the Wowza server. After reading through the SRT docs, I came across the latency flag, which I added to the Wowza streaming target and set it to zero. This reduced the delay to around 1 second. Anyway, thanks a lot for your help :)
